### PR TITLE
FDC3 ready event - Issue 241

### PR DESCRIPTION
**Pull request by sgd2z**
PR for Issue https://github.com/finos/FDC3/issues/241

### CODE REVIEW

File: docs/api/spec.md [hunk 1]
```diff
@@ -37,10 +37,23 @@ Other interfaces defined in the spec are not critical to define as concrete type
...
+  // Make some fdc3 API calls here
+}
+
+if (window.fdc3) {
+  fdc3Stuff();
+} else {
+  window.addEventListener("fdc3Ready", fdc3Stuff);
+}
+```
+
```
Comment (rikoe) [inline docs/api/spec.md:56]: Is it enough to just add this to the spec here? What about the website documentation and the typings? (Not sure if it is possible to add typings for a window-level event, just checking...)

File: docs/api/spec.md [hunk 2]
```diff
@@ -37,10 +37,23 @@ Other interfaces defined in the spec are not critical to define as concrete type
 - `DisplayMetadata`
 
 
-
 #### API Access
 The FDC3 API can be made available to an application through a number of different methods.  In the case of web applications, a Desktop Agent SHOULD provide the FDC3 API via a global accessible as _window.fdc3_. Implementors MAY additionally make the API available through modules, imports, or other means. 
 
+The global `window.fdc3` must only be available after the API is ready to use. To prevent the API from being used before it is ready, implementors may provide a global `fdc3Ready` event. Here is code demonstrating the use of the FDC3 API and the ready event:
```
Comment (rikoe) [inline docs/api/spec.md:43]: Should this be changed to "implementors SHOULD Provide", to be consistent with the rest of the spec? If we are making this addition to version 1.2, it should probably have the same mandate as `window.fdc3`?

File: docs/api/spec.md (hunk 1 above)
Comment (sgd2z) [reply docs/api/spec.md:56]: Can you point me to where the website documentation is?

Comment (rikoe) [reply docs/api/spec.md:56]: 🤔  doesn't seem there is any!

**Review by rikoe — APPROVED** [review]

**Pull request merged**